packages/lib/src/spatial-navigation/components/virtualizedGrid/SpatialNavigationVirtualizedGrid.tsx

```diff
@@ -199,17 +220,26 @@ export const SpatialNavigationVirtualizedGrid = typedMemo(
       ),
       [renderItem, numberOfColumns, rowContainerStyle],
     );
+    const renderHeaderThenRows = useCallback(
+      ({ item }: { item: GridRowType<T> | JSX.Element }) => {
+        if (React.isValidElement(item)) {
+          return item;
+        }
+        return renderRow({ item: item as GridRowType<T> });
+      },
+      [renderRow],
+    );
```

Comment:
@pierpo not really sure about this one in particular (and its little brother up there too)